### ISSUE TITLE
General Accessibility Issues for WCAG 2.1

### ISSUE THREAD

**Issue by sierralvb** [usu-access/pilot-feedback#19]
Since we're a digital accessibility office, hopefully it isn't too surprising that we did a quick analysis of the accessibility of Pilot's interface :) We can help you figure out prioritization of these later, but we do just want to bring these to your attention, as they do help make Pilot more universally usable.

This is just a big list of things that we noticed, but specific website assessment isn't our area of expertise, so if you'd like to dive a bit deeper into accessibility at some point, we would definitely recommend you do your own testing or with an agency that works on this to eventually create an [Accessibility Conformance Report](https://www.section508.gov/sell/how-to-create-acr-with-vpat/).

(We also like to err on the side of being overly helpful when explaining this stuff. If you're like "we already know that" to our explanations/suggested fixes, sorry!)

### How We Found These
Here is [the initial video](https://www.loom.com/share/4d131a587b334364ab515940cf1471eb) from when we first looked at Pilot back in July of last year using the [WAVE tool](https://wave.webaim.org/). This week, we ran through with WAVE again, and I also did a basic manual test.

### Issues
Here are a list of issues that we have found. Each individual issue can be marked off as it's completed.
- [ ] **Empty Buttons**
Both the search bar and the three dots next to documents and folders in the main content area don't have a screen reader friendly text equivalent. 
![empty button](https://github.com/usu-accessibility/Pilot-Feedback/assets/166452991/eae741db-e3e9-43c1-ab76-43e61d1c1be0)
In our experience, the easiest fix for this is to add a [sr-only style](https://uit.stanford.edu/accessibility/concepts/screen-reader-only-content), then just doing a span class="sr-only" tag within the button tag with a text equivalent in the middle. For the search bar, that text could just be "Search", and for the three dots menu, it could be something like "Options", "Menu", etc.
- [ ] **Search Bar Form Label**
The search bar doesn't have a form label, which can make it weird for screen readers. This [Stack Overflow exchange](https://stackoverflow.com/questions/43857247/making-a-search-form-accessible-label-vs-aria-label-vs-aria-labelledby) has a pretty good discussion of different fixes you can try.
- [ ] **Missing Page Language**
None of the pages have a set language. It makes sense to not automatically set a language, since not all users' wikis will be in English, but if there was a setting in the overall wiki options to set a language for wiki pages, hopefully that would be a workable fix!
- [ ] **Color Contrast**
There are four main areas where I am finding contrast issues: the Knowledge/Playlists menus, the items count in folders, updated date in pages, and then the navigation tabs in the upper left.
[Here's a Loom explaining/showing this.](https://www.loom.com/share/fd7b0003d10c4d00b27c4458aad2e7ac)
If you don't have time to watch that, here's also a typed explanation: each of these elements is just too light of a color compared to the background. If you use the WAVE tool, you can click each individual color contrast issue icon and it'll pull up a slider where you can find text colors that will fit WCAG guidelines. 
Lighter text like this can just be an issue for users with low vision or colorblindness. Here's a page if you want a more in-depth explanation of [the general issue of color contrast](https://www.usu.edu/accessibility/resources/color-contrast).
- [ ] **Keyboard Navigation Order**
While this one isn't a full-on issue, per se, I do think it's worth mentioning. Keyboard tabbing through the Pilot interface just doesn't feel like it's in the most logical order to me, as it skips the Knowledge/Playlists sections.
[Loom video explaining this.](https://www.loom.com/share/31a3a86ec88b4e81a27a42799edd3283)

**Comment by sierralvb** [comment]
@cphillipsusu I'm just seeing if tagging you on this issue will let you see it
